pkg/util/pod/pod.go

```diff
@@ -66,5 +66,6 @@ func preparePatchBytesForPodStatus(namespace, name string, uid types.UID, oldPod
 	if err != nil {
 		return nil, false, fmt.Errorf("failed to CreateTwoWayMergePatch for pod %q/%q: %v", namespace, name, err)
 	}
-	return patchBytes, bytes.Equal(patchBytes, []byte(fmt.Sprintf(`{"metadata":{"uid":%q}}`, uid))), nil
+	unchanged := bytes.Equal(patchBytes, []byte("{}")) || bytes.Equal(patchBytes, []byte(fmt.Sprintf(`{"metadata":{"uid":%q}}`, uid)))
```

Comment:
🤔 was this change necessary?

Reply:
This change is not necessary and I actually prefer to avoid making the change without protecting it by the feature gate. 

When I was adding the unit tests in gc_controller_test.go I realized that the patch requests are sent if nothing changes. 

In order to avoid the unnecessary patch I now check the result of UpdatePodCondition in gc_controller.go. 

Still, in the long run it seems sensible to make the low-level check to avoid complications of the callers code.